### PR TITLE
PRL100-293 other child details mapping

### PR DESCRIPTION
**Pull request by saranyavinayagam8**
**Before creating a pull request make sure that:**

- [ *] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ *] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### JIRA link (if applicable) ###
(https://tools.hmcts.net/jira/browse/PRLC100-293)

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/citizen/mapper/CaseDataMapper.java [hunk 1]
```diff
@@ -64,6 +65,10 @@ public CaseData buildUpdatedCaseData(CaseData caseData) throws JsonProcessingExc
             .readValue(caseData.getC100RebuildChildDetails(), C100RebuildChildDetailsElements.class);
         updateChildDetailsElementsForCaseData(caseDataBuilder, c100RebuildChildDetailsElements);
 
+        C100RebuildChildDetailsElements c100RebuildOtherChildrenDetailsElements = mapper
```
Comment (channa-patil) [inline src/main/java/uk/gov/hmcts/reform/prl/controllers/citizen/mapper/CaseDataMapper.java:68]: This needs to be revisited, I am not sure if we are using separate field for other children from the frontend

File: src/main/java/uk/gov/hmcts/reform/prl/models/c100rebuild/C100RebuildChildDetailsElements.java [hunk 2]
```diff
@@ -22,4 +22,6 @@ public class C100RebuildChildDetailsElements {
     private String childrenSubjectOfProtectionPlan;
     @JsonProperty("cd_children")
     private List<ChildDetail> childDetails;
+    @JsonProperty("cd_otherChildren")
+    private List<ChildDetail> otherChildrenDetails;
```
Comment (channa-patil) [inline src/main/java/uk/gov/hmcts/reform/prl/models/c100rebuild/C100RebuildChildDetailsElements.java:26]: @saranyavinayagam8 Are we not capturing 'cd_hasOtherChildren' yes/no field ?

**Review by channa-patil — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/citizen/mapper/CaseDataMapper.java (hunk 1 above)
Comment (saranyavinayagam8) [reply src/main/java/uk/gov/hmcts/reform/prl/controllers/citizen/mapper/CaseDataMapper.java]: changes are done

File: src/main/java/uk/gov/hmcts/reform/prl/models/c100rebuild/C100RebuildChildDetailsElements.java (hunk 2 above)
Comment (saranyavinayagam8) [reply src/main/java/uk/gov/hmcts/reform/prl/models/c100rebuild/C100RebuildChildDetailsElements.java]: added yes/no field and otherChildrenDetails in separate class

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/citizen/mapper/CaseDataOtherChildrenDetailsElementsMapper.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+
+import static org.apache.commons.lang3.ObjectUtils.isNotEmpty;
+import static uk.gov.hmcts.reform.prl.enums.YesOrNo.Yes;
+
+public class CaseDataOtherChildrenDetailsElementsMapper {
+
+    public static void updateOtherChildDetailsElementsForCaseData(CaseData.CaseDataBuilder caseDataBuilder,
+                                                                  C100RebuildOtherChildrenDetailsElements c100RebuildOtherChildrenDetailsElements) {
+
+        if ("Yes".equals(c100RebuildOtherChildrenDetailsElements.getHasOtherChildren())) {
```
Comment (channa-patil) [inline src/main/java/uk/gov/hmcts/reform/prl/controllers/citizen/mapper/CaseDataOtherChildrenDetailsElementsMapper.java:25]: @saranyavinayagam8 Can we please use 'Yes' from PrlAppsConstants ?

**Review by channa-patil — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/prl/controllers/citizen/mapper/CaseDataOtherChildrenDetailsElementsMapper.java (hunk 3 above)
Comment (saranyavinayagam8) [reply src/main/java/uk/gov/hmcts/reform/prl/controllers/citizen/mapper/CaseDataOtherChildrenDetailsElementsMapper.java]: done

**Review by channa-patil — APPROVED** [review]

**Pull request merged**